src/bots/telegram.ts

```diff
@@ -523,36 +499,49 @@ export default class TelegramBot extends BotClient {
       await this.bot.telegram.sendMessage(channel.id, text, options);
     } catch (error) {
       // Log the appropriate error
-      this.handleMessageError(error as TelegramError, channel);
-      if (retryAttempt <= MAX_SEND_MESSAGE_RETRIES) {
+      const isRetriable = this.handleMessageError(error as TelegramError, channel);
+      if (isRetriable && retryAttempt <= MAX_SEND_MESSAGE_RETRIES) {
         this.logger.warn(
-          `This was attempt ${retryAttempt} of ${MAX_SEND_MESSAGE_RETRIES} for channel ${channel.label} `,
+          `This was attempt ${retryAttempt} of ${MAX_SEND_MESSAGE_RETRIES} for channel ${channel.label}. Retrying... `,
         );
         this.sendMessage(channel, messageText, retryAttempt + 1);
         return false;
       }
 
-      // Disable the channel that failed.
       this.logger.warn(
-        `Max retries reached; Disabling failed channel ${channel.label} to avoid future errors until active again.`,
+        `Max retries reached for channel ${channel.label}, message will not be sent.`,
       );
-      this.logger.warn(`Actually, skipping disabling until it's more robust.`);
-      // channel.disabled = true;
       return false;
     }
     return true;
   }
 
-  private handleMessageError(error: TelegramError, channel: Channel) {
+  private handleMessageError(error: TelegramError, channel: Channel): boolean {
```

Comment:
Let's add a comment here, indicating what is returned from the function.
Or maybe renaming it might also make sense

Reply:
Did both, not too happy with the rename but eh